parking_lot.py

```diff
@@ -0,0 +1,201 @@
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Cannot park your car there because location {exception.location} is not available!"})
+
+
+@app.exception_handler(StartDateException)
+async def start_date_in_the_future(exception: StartDateException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Cannot part your car because date {exception.start} is invalid!"})
+
+
+@app.exception_handler(CarIdExistsException)
+async def car_id_already_exists(exception: CarIdExistsException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is already parked!"})
+
+
+@app.exception_handler(CarIdDoesNotExistException)
+async def car_id_does_not_exist(exception: CarIdDoesNotExistException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is not in the parking lot!!"})
+
+@app.exception_handler(CarIdNullException)
+async def car_id_null_exist(exception: CarIdNullException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is invalid. Car plate cannot be null!!"})
+
+
+@app.exception_handler(CarAlreadyRemovedException)
+async def car_id_already_removed(exception: CarAlreadyRemovedException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} already removed!"})
+
+@app.exception_handler(CarIdPatternException)
+async def car_id_invalid_pattern(exception: CarIdPatternException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is invalid!"})
+
+@app.get("/cars/")
+async def get_cars():
+    return PARKED_CARS
+
+
+@app.get("/cars/{car_id}", status_code=status.HTTP_200_OK)
+async def get_car(car_id: Optional[str] = None):
+    return PARKED_CARS[car_id]
+
+
+@app.post("/cars/", status_code=status.HTTP_201_CREATED)
+async def add_car(parked_car: ParkedCar):
+    if parked_car.car_id is None:
+        raise CarIdNullException(car_id=parked_car.car_id)
+
+    if parked_car.location <= 0 or parked_car.location > PARKING_LOT_CAPACITY:
+        raise InvalidLocationException(location=parked_car.location)
+
+    if parked_car.location in [x.get('location') for x in PARKED_CARS.values()]:
+        raise LocationNotAvailableException(location=parked_car.location)
+
+    start_date = parked_car.start.replace(tzinfo=None)
+    present = datetime.now()
+
+    """
+        In case the difference between the chosen date and the actual time is more
+        than the accepted time delay - by default 15 minutes - an error would be raised.
+        This is to prevent fraud or other mistakes that would decrease the fee unfairly.
+    """
+    if (start_date - present).total_seconds() > ACCEPTED_TIME_DELAY:
+        raise StartDateException(start=parked_car.start)
```

Comment:
MEDIUM ISSUE: Not clear why client (front-end) should pass datetime at all calling `add_car()`. If it is really needed why other methods don't use the same approach?